httptransport/common.go

```diff
@@ -106,3 +114,24 @@ func (a *accept) Match(mt string) bool {
 	t, s := mt[:i], mt[i+1:]
 	return a.Type == t && (a.Subtype == s || a.Subtype == "*")
 }
+
+var idPool = sync.Pool{
+	New: func() interface{} { return fnv.New64a() },
+}
+
+func withRequestID(r *http.Request) *http.Request {
+	const key = `request_id`
+	ctx := r.Context()
+	sctx := trace.SpanContextFromContext(ctx)
+	if sctx.HasTraceID() {
+		ctx = zlog.ContextWithValues(ctx, key, sctx.TraceID().String())
+	} else {
+		h := idPool.Get().(hash.Hash64)
+		h.Reset()
+		io.WriteString(h, r.RequestURI)
+		io.WriteString(h, r.RemoteAddr)
```

Comment:
Is this going to render the same hash for the same `RequestURI` `RemoteAddr` pair? Are you thinking the port will add enough entropy?

Reply:
That's my thinking. Hopefully a client isn't requesting the same URI on the same connection. Although, as I type that, perhaps  that's not good enough and we should just use a `rand` source or similar.

Reply:
I think the majority of the time it'll work :ok_hand: I just wonder if there are certain proxy situations where it's always 127.0.0.1:same_port

Reply:
Swapped it for a pool of `rand.Rand`s. That should just work in all cases.